### PR TITLE
Optimising for streaming large files

### PR DESCRIPTION
**Pull request by mryan321**
- sha dropped (we read full file for hashing)
- io.Copy is mem efficient according to my googling 😆

### CODE REVIEW

File: api/formdata.go [hunk 1]
```diff
@@ -101,12 +97,21 @@ func (f *FormDataRequest) validate(attachmentValidator FormDataValidator) (errs
 				errs = append(errs, validatorError(FileFieldKey, msg))
 			}
 
-			if data, err = ioutil.ReadAll(file); err != nil {
+			tmpZip, err := os.CreateTemp("", "s3-zip_*.zip")
```
Comment (nimitsarup) [inline api/formdata.go:105]: must remove it after use :)

**Comment by nimitsarup** [comment]
LGTM .. just 1 comment and possibly some tests to fix :)

**Review by nimitsarup — APPROVED** [review]

File: api/formdata.go (hunk 1 above)
Comment (mryan321) [reply api/formdata.go:105]: pls check but i think this is ok? https://github.com/ONSdigital/dp-interactives-api/blob/87d6cb6d056fb1745e1d0458164324073b1e67d8/api/interactives.go#L44 - so end of request

Comment (nimitsarup) [reply api/formdata.go:105]: indeed :) .. formdata.FileName is deleted at the end so all good

**Review by nimitsarup — DISMISSED** [review]

**Review by geoffmrb — APPROVED** [review]

**Pull request merged**